core/amber/src/main/protobuf/edu/uci/ics/amber/engine/architecture/rpc/controlcommands.proto

```diff
@@ -0,0 +1,251 @@
+syntax = "proto3";
+package edu.uci.ics.amber.engine.architecture.rpc;
+
+import "edu/uci/ics/amber/engine/common/virtualidentity.proto";
+import "edu/uci/ics/amber/engine/common/workflow.proto";
+import "edu/uci/ics/amber/engine/architecture/worker/statistics.proto";
+import "edu/uci/ics/amber/engine/architecture/sendsemantics/partitionings.proto";
+import "scalapb/scalapb.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/any.proto";
+
+option (scalapb.options) = {
+  scope: FILE,
+  preserve_unknown_fields: false
+  no_default_values_in_constructor: true
+};
+
+message ControlRequest {
+  oneof sealed_value {
+    PropagateChannelMarkerRequest propagateChannelMarkerRequest = 1;
+    TakeGlobalCheckpointRequest takeGlobalCheckpointRequest = 2;
+    DebugCommandRequest debugCommandRequest = 3;
+    EvaluatePythonExpressionRequest evaluatePythonExpressionRequest = 4;
+    ModifyLogicRequest modifyLogicRequest = 5;
+    RetryWorkflowRequest retryWorkflowRequest = 6;
+    ConsoleMessageTriggeredRequest consoleMessageTriggeredRequest = 8;
+    PortCompletedRequest portCompletedRequest = 9;
+    WorkerStateUpdatedRequest workerStateUpdatedRequest = 13;
+    LinkWorkersRequest linkWorkersRequest = 15;
+    Ping ping = 16;
+    Pong pong = 17;
+    Nested nested = 18;
+    Pass pass = 19;
+    ErrorCommand errorCommand = 20;
+    Recursion recursion = 21;
+    Collect collect = 22;
+    GenerateNumber generateNumber = 23;
+    MultiCall multiCall = 24;
+    Chain chain = 25;
+    AddInputChannelRequest addInputChannelRequest = 26;
+    AddPartitioningRequest addPartitioningRequest = 27;
+    AssignPortRequest assignPortRequest = 28;
+    FinalizeCheckpointRequest finalizeCheckpointRequest = 29;
+    InitializeExecutorRequest initializeExecutorRequest = 30;
+    UpdateExecutorRequest updateExecutorRequest = 31;
+    EmptyRequest emptyRequest = 33;
+    PrepareCheckpointRequest prepareCheckpointRequest = 34;
+    QueryStatisticsRequest queryStatisticsRequest = 35;
```

Comment:
to help future maintenance and avoid massive line changes, can we do the following:
1. Sort the requests in a way, either by alphabetical order, or group them by functionality.
2. Leave some buffer numbers for possible future new Requests defintion. Currently the numbers are continued, to add a request in the middle of the sequence would cause all other number to shift.